synapse/util/async_helpers.py

```diff
@@ -268,7 +269,7 @@ def __init__(
         if not clock:
             from twisted.internet import reactor
 
-            clock = Clock(reactor)  # type: ignore[arg-type]
+            clock = Clock(cast(IReactorTime, reactor))
```

Comment:
Does it work if you add a `assert isinstance(reactor, IReactorTime)` above?

Reply:
Mypy likes it, but `assert isinstance(reactor, IReactorTime)` gives an AssertionError at runtime :(.

EPollReactor has `ReactorBase` in its ancestry which implements `IReactorTime`, so it seems like doing `assert isinstance(reactor, ReactorBase)` is actually what we want (and appears to work).